networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/gossip/AttestationSubnetSubscriptions.java

```diff
@@ -0,0 +1,96 @@
+ */
+
+package tech.pegasys.artemis.networking.eth2.gossip;
+
+import static tech.pegasys.artemis.util.config.Constants.ATTESTATION_SUBNET_COUNT;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.primitives.UnsignedLong;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import tech.pegasys.artemis.networking.eth2.gossip.topics.AttestationTopicHandler;
+import tech.pegasys.artemis.networking.p2p.gossip.GossipNetwork;
+import tech.pegasys.artemis.networking.p2p.gossip.TopicChannel;
+import tech.pegasys.artemis.storage.client.RecentChainData;
+
+public class AttestationSubnetSubscriptions implements AutoCloseable {
+  private final GossipNetwork gossipNetwork;
+  private final RecentChainData recentChainData;
+  private final EventBus eventBus;
+
+  private final Map<UnsignedLong, Set<UnsignedLong>> subnetIdToCommittees = new HashMap<>();
+  private final Map<UnsignedLong, TopicChannel> subnetIdToTopicChannel = new HashMap<>();
+
+  public AttestationSubnetSubscriptions(
+      final GossipNetwork gossipNetwork,
+      final RecentChainData recentChainData,
+      final EventBus eventBus) {
+    this.gossipNetwork = gossipNetwork;
+    this.recentChainData = recentChainData;
+    this.eventBus = eventBus;
+  }
+
+  public synchronized Optional<TopicChannel> getChannel(final UnsignedLong committeeIndex) {
+    final UnsignedLong subnetId = committeeIndexToSubnetId(committeeIndex);
+    return Optional.ofNullable(subnetIdToTopicChannel.get(subnetId));
+  }
+
+  public synchronized void subscribeToCommitteeTopic(final UnsignedLong committeeIndex) {
+    final UnsignedLong subnetId = committeeIndexToSubnetId(committeeIndex);
+    final Set<UnsignedLong> subscribedCommittees =
+        subnetIdToCommittees.computeIfAbsent(subnetId, __ -> new HashSet<>());
+    subscribedCommittees.add(committeeIndex);
+    subnetIdToTopicChannel.computeIfAbsent(subnetId, this::createChannelForSubnetId);
+  }
+
+  public synchronized void unsubscribeFromCommitteeTopic(final UnsignedLong committeeIndex) {
+    final UnsignedLong subnetId = committeeIndexToSubnetId(committeeIndex);
+    final Set<UnsignedLong> committees =
+        subnetIdToCommittees.getOrDefault(subnetId, Collections.emptySet());
+    committees.remove(committeeIndex);
+    if (!committees.isEmpty()) {
+      // We still have some subscribers, don't actually unsubscribe
+      return;
+    }
+    subnetIdToCommittees.remove(subnetId);
+    final TopicChannel topicChannel = subnetIdToTopicChannel.remove(subnetId);
+    if (topicChannel != null) {
+      topicChannel.close();
+    }
+  }
+
+  private UnsignedLong committeeIndexToSubnetId(final UnsignedLong committeeIndex) {
+    final UnsignedLong subnetId =
+        committeeIndex.mod(UnsignedLong.valueOf(ATTESTATION_SUBNET_COUNT));
+    return subnetId;
```

Comment:
nit: Inline the variable
```suggestion
    return committeeIndex.mod(UnsignedLong.valueOf(ATTESTATION_SUBNET_COUNT));
```

Reply:
Done.